tests/templates/test_state_preparations.py

```diff
@@ -359,6 +392,157 @@ def circuit(state_vector):
             state_vector = np.array([0, 2, 0, 0])
             circuit(state_vector)
 
+    # fmt: off
+    @pytest.mark.parametrize("state_vector, n_wires", [
+        ([.5,.5,.5,.5],2),
+        ([1, 0, 0, 0], 2),
+        ([0, 1, 0, 0], 2),
+        ([0, 0, 0, 1], 2),
+        ([0, 1, 0, 0, 0, 0, 0, 0], 3),
+        ([0, 0, 0, 0, 1, 0, 0, 0], 3),
+        ([2/3, 0, 0, 0, 1/3, 0, 0, 2/3], 3),
+        ([1/2, 0, 0, 0, 1/2, 1/2, 1/2, 0], 3),
+        ([1/3, 0, 0, 0, 2/3, 2/3, 0, 0], 3),
+        ([2/3, 0, 0, 0, 1/3, 0, 0, 2/3], 3),
+    ])
+    # fmt: on
+    def test_RZ_skipped(self, state_vector, n_wires):
+        """Tests whether the cascade of RZ gates is skipped for real-valued states"""
+
+        n_CNOT = 2 ** n_wires - 2
+
+        dev = qml.device("default.qubit", wires=n_wires)
+
+        @qml.qnode(dev)
+        def circuit(state_vector):
+            MottonenStatePreparation(state_vector, wires=range(n_wires))
+            return qml.expval(qml.PauliX(wires=0))
+
+        # when the RZ cascade is skipped, CNOT gates should only be those required for RY cascade
+        circuit(state_vector)
+
+        assert circuit.qtape.get_resources()["CNOT"] == n_CNOT
+
+    @pytest.mark.parametrize(
```

Comment:
@DSGuala  These tests are fine, but it's important to keep tests lightweight because they all have to be run on every new commit on every PR in the repo!

Reply:
Thanks, that's a great point.